src/main/kotlin/org/misarch/discount/service/DiscountService.kt

```diff
@@ -294,6 +292,29 @@ class DiscountService(
         }
     }
 
+    /**
+     * Finds all discounts that are applicable to a user, and a product variant and a list of coupons.
+     * Does NOT check if the user can use the discounts with the amount of items.
+     *
+     * @param productVariantInput defines the product variant, the count, and the coupons to check
+     * @param input defines the id of the user for which to check the coupons and the order amount
+     * @return The applicable discounts for the product variant, count and user
+     */
+    private suspend fun findApplicableDiscountsForProductVariant(
+        productVariantInput: FindApplicableDiscountsProductVariantInput,
+        input: FindApplicableDiscountsInput
+    ): MutableList<DiscountEntity> {
```

Comment:
Why `mutable`?
Where are you changing the list?

Reply:
nowhere, was the result of a refactoring (returned from java, thus Kotlin assumes mutable)
changed with f025be506bd52dc4b316094cf8d8933083fd3411